src/recipe/recipe.repository.ts

```diff
@@ -56,10 +56,10 @@ export class RecipeRepository {
     return this.prisma.recipe.findMany();
   }
 
-  getAllPublicRecipes(): Promise<Recipe[]> {
+  getAllPublicRecipes(authorId: number): Promise<Recipe[]> {
```

Comment:
Naming is kind of misleading because here we are getting both public recipes and private recipes but belonging to the `authorId`. It should be named better - I don't have a good idea right now but perhaps something along `getAllAvailableRecipesForUser(userId)` or `getAllRecipesTheUserHasAccessTo(userId)` or `getAllPublicAndOwnerRecipes(ownerId)` or something else.